README.md

```diff
@@ -1,24 +1,161 @@
-to do that we need to integrate route variables into our routes for a post /posts/1 can't be hardcoded.
+If you are comfortable with how static requests you will pick up on dynamic requests quickly. An example scenario for a dynamic request would be where a user navigates to a page in the application such as ```localhost:3000/posts/42``` and is displayed a post with the id of 42. Behind the scenes what is happening is below:
 
-how to draw a dynamic route, :variable syntax and how that is arbitrary but will correspond to params.
+1. The ```route.rb``` file takes in the request and processes it like normal, except this time it also parses the ```42``` parameter and passes it to the posts' controller.
 
-how to access the dynamic route variable in a controller action and param how to use that data in find Post
+2. From that point the controller parses the ```42``` parameter and runs a query on the Post model, storing the result in an instance variable.
 
-passing an instance variable to a view
+3. Lastly the controller passes the instance variable to the associated view which renders that specific post record details to the client.
+
+
+## Code Implementation
+
+In order to setup a dynamic request feature we will start by writing a test to verify that the page exists (you can ignore the FactoryGirl calls, I'm simply using it to create a new post so we have an object to work with):
+
+```ruby
+# spec/features/post_spec.rb
+
+require 'rails_helper'
+
+feature 'navigate to post pages' do
+  subject { FactoryGirl.create(:post) }
+
+  scenario 'on the show page' do
+    visit "/posts/#{subject.id}"
+    expect(page.status_code).to eq(200)
+  end
+
+end
+```
+
+Running ```bundle exec rspec``` gives us an expected error of: ```ActionController::RoutingError: No route matches [GET] "/posts/1"```. To correct this error let's draw a route that maps to a show action in the posts' controller:
+
+```ruby
+get 'posts/:id', to: 'posts#show'
+```
+
+Here you will notice something that's different from the static route, the ```/:id``` tells the routing system that this route can receive a parameter, and that parameter will be passed to the controller's show action. With this route in place let's run our tests again, you will see we get a different failure this time that says: ```AbstractController::ActionNotFound: The action 'show' could not be found for PostsController```. This means that we need to create a corresponding ```show``` action in the posts' controller, let's get this failure fixed with the code below:
+
+```ruby
+# app/controllers/posts_controller.rb
+
+class PostsController < ApplicationController
+  def show
+  end
+end
+```
+
+Running the tests now you will get a failure saying that that we have a missing view template, let's fix that by creating a view file that is mapped to our ```show``` action by running the following terminal command (we also need to create a posts directory): ```mkdir app/views/posts && touch app/views/posts/show.html.erb```
```

Comment:
also just tell them to create a posts folder in the view folder and create a `show.html.erb` file in there